public/app/features/serviceaccounts/ServiceAccountPage.tsx

```diff
@@ -70,7 +70,9 @@ export const ServiceAccountPageUnconnected = ({
 
   const serviceAccountId = parseInt(match.params.id, 10);
   const tokenActionsDisabled =
-    !contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite) || serviceAccount.isDisabled;
+    serviceAccount.isManaged &&
+    contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite) &&
+    !serviceAccount.isDisabled;
```

Comment:
```suggestion
    !contextSrv.hasPermission(AccessControlAction.ServiceAccountsWrite) ||
    serviceAccount.isManaged ||
    !serviceAccount.isDisabled;
```

Reply:
Changed as requested

Reply:
@gamab this suggestion is not the same as the original code nor my changes.

My code changes do 2 things
1. use de morgan law to invert the boolean expression from `OR` to `AND`
2. append the `isManaged` with `AND`

Your code suggestion is not the same because:
1. you inverted `serviceAccount.isDisabled` but not changing anything else, the `OR` or the permission check
2. append the `isManaged` with `OR`

this will result in the following test failing:
```
Should call API with proper params when delete service account token
```

Reply:
I changed the code from `AND` to `OR` for all 3 boolean expressions but kept `serviceAccount.isDisabled` without negating it